source/dagger/gameplay/ping_pong/player_scores.cpp

```diff
@@ -48,8 +49,7 @@ void PlayerScoresSystem::Run()
 
     if (ballOnField == 0)
     {
-        auto& reg = Engine::Registry();
-        auto sys = Engine::GetDefaultResource<PingPongBallSystem>();
-        sys->CreatePingPongBall(s_TileSize, ColorRGBA(1, 1, 1, 1), { rand()%10 + 4,rand()%10 + 4,0 },   { 0,rand()%(s_FieldHeight / 2),0 });
+        // TODO: set speed to be random in both directions
+        CreatePingPongBall(s_TileSize, ColorRGBA(1, 1, 1, 1), { rand() % 10 - 4, rand() % 10 + 4,0 }, { 0,rand() % (s_FieldHeight / 2),0 });
```

Comment:
try to fix this todo pls

Reply:
It should be working now : ) it can go both left and right initially

Reply:
What happens if `rand()` returns `4`?

Reply:
Right, it goes only vertically, fixed that